examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+{
+  logString = "Message received: Topic: ";
+  LogInfo(logString + mqttClient.messageTopic() + ", Length: " + messageSize);
+  LogInfo("Message: ");
+
+  while (mqttClient.available()) 
+  {
+    Serial.print((char)mqttClient.read());
+  }
+  Serial.println();
+}
+
+/*
+ * sendTelemetry:
+ * The Azure IoT SDK for C client creates the MQTT topic to publish a telemetry message.
+ * The MQTT client creates and sends the telemetry mesage on the topic.
+ */
+static void sendTelemetry() 
+{
+  digitalWrite(LED_PIN, HIGH);
+  LogInfo("Arduino Nano RP2040 Connect sending telemetry . . . ");
+
+  int result = az_iot_hub_client_telemetry_get_publish_topic(
+      &azIoTHubClient, NULL, telemetryTopic, sizeof(telemetryTopic), NULL);
+  if (az_result_failed(result)) 
+  {
+    logString = "Failed to get telemetry publish topic. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  mqttClient.beginMessage(telemetryTopic);
+  mqttClient.print(generateTelemetry());
+  mqttClient.endMessage();
+
+  LogInfo("Telemetry sent.");
+  delay(100);
+  digitalWrite(LED_PIN, LOW);
+}
+
+/*
+ * generateTelemetry:
+ * Simulated telemetry.  
+ * In your application, this function should retrieve real telemetry data from the device and format
+ * it as needed.
+ */
+static char* generateTelemetry() 
+{
+  String payloadStart = "{ \"msgCount\": ";
+  telemetryPayload =  payloadStart + telemetrySendCount + " }";
+  telemetrySendCount++;
+
+  return (char*)telemetryPayload.c_str();
+}
+
+/*************************************/
+/*    SAS Token related functions    */
+/*************************************/
+
+/*
+ * generateMQTTPassword:
+ * The MQTT password is the generated SAS token. The process is:
+ *    1. Get the SAS token expiration time from the provided value. (Default 60 minutes).
+ *    2. Azure IoT SDK for C creates the SAS signature from this expiration time.
+ *    3. Sign and encode the SAS signature.
+ *    4. Azure IoT SDK for C creates the MQTT Password from the expiration time and the encoded,
+ *       signed SAS signature.
+ */
+static void generateMQTTPassword() 
+{
+  int result;
+
+  uint64_t sasTokenDuration = 0;
+  uint8_t signature[BUFFER_LENGTH_SAS_SIGNATURE] = {0};
+  az_span signatureAzSpan = AZ_SPAN_FROM_BUFFER(signature);
+  uint8_t encodedSignedSignature[BUFFER_LENGTH_SAS_ENCODED_SIGNED_SIGNATURE] = {0};
+  size_t encodedSignedSignatureLength = 0;
+
+  // Get the signature. It will be signed later with the decoded device key.
+  // To change the sas token duration, see IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES in iot_configs.h
+  sasTokenDuration = getSASTokenExpirationTime(IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES);
+  result = az_iot_hub_client_sas_get_signature(
+      &azIoTHubClient, sasTokenDuration, signatureAzSpan, &signatureAzSpan);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the signature for SAS Token. Return code: ";
+    LogError(logString + result);
+    exit(result);

```

Comment:
Same comment for exit. The correct implementation should be getting into an infinite loop, which is then debuggable in platforms that so support it.